chains/evm/contracts/offRamp/OffRamp.sol

```diff
@@ -706,10 +708,8 @@ contract OffRamp is ITypeAndVersion, Ownable2StepMsgSender {
       sourceDenominatedAmount: sourceTokenAmount.amount,
       localToken: localToken,
       remoteChainSelector: sourceChainSelector,
-      // This re-encodes the address as bytes, but now with the zero prefix to make it 32 bytes long.
-      // We have to cast it to an address to ensure the bytes are padded on the left with zeros, bytes objects get
-      // padded on the right.
-      sourcePoolAddress: abi.encode(address(bytes20(sourceTokenAmount.sourcePoolAddress))),
+      // If the source pool is from an evm, the bytes are already abi encoded.
```

Comment:
Maybe note a more general rule: the source chain has encoded this in the expected format. That's true for all chains, not just EVMs.